main/drivers/CANDriver.h

```diff
@@ -0,0 +1,123 @@
+/******************************************************************************
+ * CANDriver.h
+ *
+ * The CANDriver serves as a high level interface for receiving CAN messages.
+ * The driver will setup multiple CAN chips on a single SPI bus and handle
+ * automatically receiving messages in a separate task. All received messages
+ * are held in a single queue with each message designating the bus (receiver
+ * chip) of origin.
+ *****************************************************************************/
+#ifndef CAN_DRIVER_H
+#define CAN_DRIVER_H
+
+// Imports
+#include <driver/gpio.h>
+#include <driver/spi_master.h>
+
+#include <freertos/event_groups.h>
+#include <freertos/message_buffer.h>
+#include <freertos/semphr.h>
+
+#include "GUB_Conf.h"
+#include "drivers/Driver_MCP251863.h"
+
+// Global Defines
+#define CAN_FIFO_COUNT 2
+
+#ifndef CAN_MAX_MESSAGE_SIZE
```

Comment:
The minimum is 8, but it can be larger. It is just a safety check since it comes from the config file and not all can be trused with the power of macros. If you want I can make the compiler error out instead.

Reply:
Oh, I see, you're just making sure the max is never less than 8. Makes sense